### PR TITLE
Add sort parameter to checkout fields to aid with sorting per locale.

### LINKED ISSUE #12553: address-i18n.js causing address order problems for JP users
address-i18n.js L45 ( Re-order postcode / city ) forces the postcode to come either before the city or after the state depending on 'postcode_before_city' locale setting.

For Japanese users the postcode should come before both the city and the state.
The order should be Postcode -> State -> City

However the JS makes this impossible without a core hack.

I would suggest moving any address re-ordering to occur in PHP before filters are applied so users have the chance to override.

### PR DESCRIPTION
**Pull request by mikejolley**
Instead of having a postcode before city arg, add a new sort arg to all checkout fields so they can be sorted dynamically on display per locale.

Closes #12553

### CODE REVIEW

File: assets/js/frontend/address-i18n.js [hunk 1]
```diff
@@ -113,9 +105,37 @@ jQuery( function( $ ) {
...
+					}
 				}
 
 			});
-		});
 
+			var fieldsets = $('.woocommerce-billing-fields, .woocommerce-shipping-fields, .woocommerce-address-fields');
+
+			fieldsets.each( function( index, fieldset ) {
+				var rows = $( fieldset ).find('.form-row');
```
Comment (claudiosanches) [inline assets/js/frontend/address-i18n.js]: Missing one space: `find('.form-row' );`.

File: assets/js/frontend/address-i18n.js [hunk 2]
```diff
@@ -113,9 +105,37 @@ jQuery( function( $ ) {
...
+
+				rows.sort(function( a, b ){
```
Comment (claudiosanches) [inline assets/js/frontend/address-i18n.js]: Missing spaces `rows.sort( function( a, b ) {`

File: assets/js/frontend/address-i18n.js [hunk 3]
```diff
@@ -113,9 +105,37 @@ jQuery( function( $ ) {
...
+					var asort = $(a).data('sort'),
```
Comment (claudiosanches) [inline assets/js/frontend/address-i18n.js]: Missing spaces here too: `$( a ).data( 'sort' ),`

File: assets/js/frontend/address-i18n.js [hunk 4]
```diff
@@ -113,9 +105,37 @@ jQuery( function( $ ) {
...
+						bsort = $(b).data('sort');
```
Comment (claudiosanches) [inline assets/js/frontend/address-i18n.js]: And here: `$( b ).data( 'sort' );`

File: includes/class-wc-countries.php [hunk 5]
```diff
@@ -403,53 +403,50 @@ public function country_dropdown_options( $selected_country = '', $selected_stat
...
+	 *
+	 * These define how addresses are formatted for display in various countries.
+	 *
 	 * @return array
 	 */
 	public function get_address_formats() {
 		if ( empty( $this->address_formats ) ) {
-
-			// Common formats
-			$postcode_before_city = "{company}\n{name}\n{address_1}\n{address_2}\n{postcode} {city}\n{country}";
```
Comment (claudiosanches) [inline includes/class-wc-countries.php]: Some reason to avoid "DRY"?

**Review by claudiosanches — CHANGES_REQUESTED** [review]

File: includes/class-wc-countries.php (hunk 5 above)
Comment (mikejolley) [inline includes/class-wc-countries.php]: postcode_before_city is no longer relevant so I didn't want it here.

File: assets/js/frontend/address-i18n.js [hunk 6]
```diff
@@ -130,7 +130,7 @@ jQuery( function( $ ) {
 				} );
 
 				// Sort the fields.
-				rows.sort(function( a, b ){
+				rows.sort( function( a, b ) {
 					var asort = $( a ).data('priority'),
```
Comment (claudiosanches) [inline assets/js/frontend/address-i18n.js]: Missing spacing here too :P

**Review by claudiosanches — CHANGES_REQUESTED** [review]

**Pull request merged**